internal/controller/resource_manager.go

```diff
@@ -112,9 +88,9 @@ func (r *ResourceManager) DyanmicListAndWatchResources(ctx context.Context, canc
 }
 
 // DynamicListResources lists a specifed resource dynamically and sends down the current gRPC stream.
-func (r *ResourceManager) DynamicListResources(ctx context.Context, resource string) (string, CacheManager, error) {
+func (r *ResourceManager) DynamicListResources(ctx context.Context, resource string, apiGroup string) (string, CacheManager, error) {
 	cm := CacheManager{cache: make(map[string][32]byte)}
-	objGVR := schema.GroupVersionResource{Group: resourceAPIGroupMap[resource], Version: "v1", Resource: resource}
+	objGVR := schema.GroupVersionResource{Group: apiGroup, Version: "v1", Resource: resource}
```

Comment:
Seeing this, I think we shouldn't hardcode the `"v1"` version, just in case.
A cleaner design would be to make `resourceAPIGroupMap` a list of `schema.GroupVersionResource` values. You could then pass around those `schema.GroupVersionResource` values instead of API group + resource type.

Do this in a separate PR if you want.
This is not blocking the PR.

Reply:
Good point. Will make a note of this!